src/data/FetchHttpClientRepository.ts

```diff
@@ -83,6 +83,16 @@ export class FetchHttpClientRepository implements HttpClientRepository {
     getMockAdapter(): MockAdapter {
         throw new Error("Not implemented");
     }
+
+    private getAuthHeaders(auth?: Credentials, personalToken?: PATToken): Record<string, string> {
+        if (auth) {
```

Comment:
So authentication will always be user/pass -> personalToken -> nothing. This, in the types world, is a union. Does it make sense to have this?

```typescript
type Auth =
    | { type?: "basicAuth"; username: string; password: string }
    | { type: "personalToken"; token: string };
```

(The idea of `type?` is to support the default auth without changes to clients)